### PR TITLE
add mini-frame-border-color.

### PR DESCRIPTION
**Pull request by jixiuf**

### CODE REVIEW

File: mini-frame.el [hunk 1]
```diff
@@ -92,6 +92,10 @@ to determine background color of mini frame."
 Called if `mini-frame-show-parameters' doesn't specify background color."
   :type 'function)
 
+(defcustom mini-frame-border-color "gray50"
```
Comment (muffinmad) [inline mini-frame.el:95]: Let's name it `mini-frame-internal-border-color`. Also please make in `nil` by default.

File: mini-frame.el [hunk 2]
```diff
@@ -92,6 +92,10 @@ to determine background color of mini frame."
...
+  "MiniFrame border color."
+  :type 'string)
```
Comment (muffinmad) [inline mini-frame.el:97]: Use `color` instead of `string` for type.

File: mini-frame.el [hunk 3]
```diff
@@ -237,6 +241,8 @@ ALIST is passed to `window--display-buffer'."
                                 show-parameters)))
       (set-face-background 'fringe nil mini-frame-completions-frame))
     (modify-frame-parameters mini-frame-completions-frame show-parameters)
+    (when mini-frame-border-color
+      (set-face-background 'internal-border mini-frame-border-color mini-frame-completions-frame))
```
Comment (muffinmad) [inline mini-frame.el:245]: Can this be done once the frame is created? Like modifying `fringe` background.

File: mini-frame.el [hunk 4]
```diff
@@ -272,6 +278,8 @@ ALIST is passed to `window--display-buffer'."
                                   show-parameters)))
         (set-face-background 'fringe nil mini-frame-frame)))
     (modify-frame-parameters mini-frame-frame show-parameters)
+    (when mini-frame-border-color
+      (set-face-background 'internal-border mini-frame-border-color mini-frame-frame))
```
Comment (muffinmad) [inline mini-frame.el:282]: Same here. No need to set border color every time the frame is shown.

**Review by muffinmad — CHANGES_REQUESTED** [review]
Thanks! Few comments that should be addressed before this gets merged.

File: mini-frame.el (hunk 1 above)
Comment (jixiuf) [reply mini-frame.el]: done

File: mini-frame.el (hunk 2 above)
Comment (jixiuf) [reply mini-frame.el]: done

File: mini-frame.el (hunk 4 above)
Comment (jixiuf) [reply mini-frame.el]: I tried, but it seems does not work, I am using 28.0.50 on macOS

Comment (muffinmad) [reply mini-frame.el]: Mini-frame is also created here: https://github.com/muffinmad/emacs-mini-frame/blob/1fd7fcac0a5c6ba5bfdb9be00f9594fd8ffa90fd/mini-frame.el#L367

Comment (jixiuf) [reply mini-frame.el]: Same result, still doesn't work . it only work when the child frame first show.

Comment (muffinmad) [reply mini-frame.el]: Did you set the `mini-frame-border-color` before enabling the `mini-frame-mode`?

File: mini-frame.el [hunk 5]
```diff
@@ -92,6 +92,10 @@ to determine background color of mini frame."
 Called if `mini-frame-show-parameters' doesn't specify background color."
   :type 'function)
 
+(defcustom mini-frame-internal-border-color nil
+  "MiniFrame internal border color."
+  :type 'color)
```
Comment (muffinmad) [inline mini-frame.el:97]: `nil` and `:type 'color` will lead to this:

Знімок екрана 2020-10-06 о 17 29 06

This defcustom must have the type like this:

```elisp
   :type '(choice (const :tag "None" nil)
                  (color :tag "Color")))
```

File: mini-frame.el (hunk 4 above)
Comment (jixiuf) [reply mini-frame.el]: yes 
```
(setq mini-frame-resize-max-height max-mini-window-height)
(setq mini-frame-completions-focus 'minibuffer)
(setq mini-frame-internal-border-color "gray70")
(when (require 'mini-frame nil t)
  (add-to-list 'mini-frame-ignore-commands 'dired-narrow)
  (setq mini-frame-show-parameters
        '((top . 0.4)
          (width . 1.0)
          (min-height .  5)
          (height . 15)
          (minibuffer-exit . t)     
          (alpha . 100)
          (left-fringe . 10 )
          (cursor-color . "Yellow")
          (background-color . "#2a3a4a")    
          (background-mode 'light)
     
          ))

  (mini-frame-mode 1))


```

File: mini-frame.el (hunk 5 above)
Comment (jixiuf) [reply mini-frame.el]: fixed.

File: mini-frame.el (hunk 4 above)
Comment (jixiuf) [reply mini-frame.el]: I print the value of  `(face-background 'internal-border mini-frame-frame )` before and after `(modify-frame-parameters mini-frame-frame show-parameters)`,
it seems that `modify-frame-parameters` changed that value from my `gray70` to `nil`

Comment (muffinmad) [reply mini-frame.el]: You are right. Filled [bug report](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=43837) with minimal steps to reproduce this issue.

BTW there are typo in your code snippet. Replace `(background-mode 'light)` with `(background-mode . light)` ;)

Comment (muffinmad) [reply mini-frame.el]: The issue is [fixed now](http://git.savannah.gnu.org/cgit/emacs.git/commit/?id=bc2b63876b669ede07c93bc5694030646d9fbd36) so we can avoid calling `set-face-background` every time the mini frame is shown.

Comment (jixiuf) [reply mini-frame.el]: ok I make a force push .Will test it after emacs compiled

Comment (jixiuf) [reply mini-frame.el]: it works

**Review by muffinmad — APPROVED** [review]

**Comment by muffinmad** [comment]
Merged. Thanks!

**Pull request merged**